src/component/feature/FeatureStrategy/FeatureStrategyForm/FeatureStrategyForm.tsx

```diff
@@ -45,44 +53,78 @@ export const FeatureStrategyForm = ({
     setStrategy,
     segments,
     setSegments,
+    errors,
 }: IFeatureStrategyFormProps) => {
     const { classes: styles } = useStyles();
     const [showProdGuard, setShowProdGuard] = useState(false);
     const hasValidConstraints = useConstraintsValidation(strategy.constraints);
     const enableProdGuard = useFeatureStrategyProdGuard(feature, environmentId);
     const { hasAccess } = useContext(AccessContext);
+    const { strategies } = useStrategies();
     const navigate = useNavigate();
 
+    const strategyDefinition = strategies.find(definition => {
+        return definition.name === strategy.name;
+    });
+
     const {
         uiConfig,
         error: uiConfigError,
         loading: uiConfigLoading,
     } = useUiConfig();
 
+    if (uiConfigError) {
+        throw uiConfigError;
+    }
+
+    if (uiConfigLoading || !strategyDefinition) {
+        return null;
+    }
+
+    const findParameterDefinition = (name: string): IStrategyParameter => {
+        return strategyDefinition.parameters.find(p => p.name === name)!;
+    };
+
+    const validateParameter = (
+        name: string,
+        value: IFeatureStrategyParameters[string]
+    ): boolean => {
+        const parameterValueError = validateParameterValue(
+            findParameterDefinition(name),
+            value
+        );
+        if (parameterValueError) {
+            errors.set(name, parameterValueError);
+            return false;
+        } else {
+            errors.remove(name);
+            return true;
+        }
+    };
+
+    const validateAllParameters = (): boolean => {
+        return strategyDefinition.parameters
+            .map(p => validateParameter(p.name, strategy.parameters?.[p.name]))
```

Comment:
Please update this to full variable name so that scanning the code become easier.